sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2690,6 +2690,18 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val GROUPING_WITH_UNION = buildConf("spark.sql.optimizer.grouping")
+    .doc("When true, enable grouping with union.")
+    .booleanConf
+    .createWithDefault(false)
```

Comment:
Is this config needed?

Reply:
yes. The default spark.sql.optimizer.projections.size=4,but if your sql's projections excceed 4,and you still don't want to turn on this function. you can set spark.sql.optimizer.grouping=false